### PR TITLE
method to add Elastic axis to MatrixJob

### PR DESCRIPTION
**Pull request by odorovic**
Hi,

This PR contains the code, unit tests and job reference update. I hope that it's in order.

Cheers,
Aleks

MatrixJob does not have a convinience method to add elastic axis. Define
it.

--Define elastic method under Axis node.
--Add a basic unit test.
--Update job reference

[JENKINS-29866]

### CODE REVIEW

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/AxisContext.groovy [hunk 1]
```diff
@@ -48,6 +48,16 @@ class AxisContext extends AbstractExtensibleContext {
         simpleAxis('JDK', 'jdk', axisValues)
     }
 
+    void elastic(String axisName, String axisValues, boolean ignoreOffline = true) {
+        NodeBuilder nodeBuilder = new NodeBuilder()
+
+        Node node = nodeBuilder.'org.jenkinsci.plugins.elasticaxisplugin.ElasticAxis'(plugin: 'elastic-axis@1.2')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/AxisContext.groovy]: Remove the plugin attribute, the DSL does not generate them.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/AxisContext.groovy [hunk 2]
```diff
@@ -48,6 +48,16 @@ class AxisContext extends AbstractExtensibleContext {
         simpleAxis('JDK', 'jdk', axisValues)
     }
 
+    void elastic(String axisName, String axisValues, boolean ignoreOffline = true) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/AxisContext.groovy]: Add a `@RequiresPlugin` annotation including a minimum version.

File: docs/Job-reference.md [hunk 3]
```diff
@@ -3433,6 +3434,23 @@ matrixJob('example') {
...
+This is useful for distributing tests, as well as running jobs related to slave setup.
+
+```groovy
+matrixJob {
+    axes {
+        elastic(String name, String values, boolean ignoreOffline=true)
+    }
+}
+```
+
```
Comment (daspilker) [inline docs/Job-reference.md]: add an example

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/jobs/MatrixJobSpec.groovy [hunk 4]
```diff
@@ -76,6 +76,18 @@ class MatrixJobSpec extends Specification {
         job.node.axes[0].children()[0].name() == 'hudson.matrix.LabelAxis'
     }
 
+    def 'can set leastic axis'() {
+        when:
+        job.axes {
+            elastic('AXIS NAME1', 'label1')
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/jobs/MatrixJobSpec.groovy]: test all options

**Comment by daspilker** [comment]
Please rebase to master, adapt to the new documentation style and address the inline comments. Otherwise I will close this pull request.

**Comment by odorovic** [comment]
Hi,

Sorry about this. I solved my problem without the elastic axis so I forgot about this. Please do close and I'll ask for a pull once I have some time available to fix everything.

**Pull request closed without merging**